### PR TITLE
Fix: Cursors over UI elements should indicate clickable or text entry

### PR DESCRIPTION
**Pull request by belcherj**
### Fix
Cursors in the UI do not use the proper type to indicate when elements are clickable or editable. This PR fixes those issue across the app.

### Test
1. Run the branch
2. Step through app hovering over anything that is clickable and verify that the cursor changes to "pointer"
3. Hover over anything that allows text entry and verify that the cursor changes to "text"

### Review
Only one developer is required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated with:

- Hovering over a clickable or editable UI element now show the correct cursor for its type

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,10 +3,12 @@
 ## Future Release
 
 - Updated Log in and Sign up form to match current styling
+- Fixed cursors to use the proper type
```
Comment (dmsnell) [inline CHANGELOG.md]: Can we provide a stronger explanation of this? Where will people expect to find changes? Why were they wrong?

Something simple would probably go a long way…

```
Changed cursor over non-editable items to pointer
```

What were the cursors even before-hand?

**Comment by dmsnell** [comment]
I'm not sure what to expect. Are there any before/after screenshots? I don't particularly know what "the proper type" means or how I would know that a cursor is the wrong type. Is there a way we could provide any more description in here?

**Comment by codebykat** [comment]
@dmsnell There were essentially no pointer cursors over any clickable elements. I'll 👀

File: CHANGELOG.md (hunk 1 above)
Comment (codebykat) [reply CHANGELOG.md]: Something like this, maybe?

> Hovering over a clickable or editable UI element now show the correct cursor for its type

**Comment by belcherj** [comment]
> The note list on the sidebar (to switch notes, or to pin/unpin a note -- should be pointer, is default)

This should have resolved that https://github.com/Automattic/simplenote-electron/pull/1573/files#diff-b4aef3f86a137f3edaf5a4e2fbb2a23fR104 Do you not see the pointer

> Tags at the bottom of a note, when it has tags (should be pointer, is text cursor)

> "Add a tag..." at the bottom of a note (should be text cursor, is default -- only in Firefox, it's correct in Safari)

Made Tags that are on a note pointer since clicking on them removes from note. In the UI spring cleaning, this interaction will be cleaned up. Where you enter new tags I changed the cursor to text.

**Comment by codebykat** [comment]
I do not see the pointer over the notes sidebar, over the pin/unpin or over the note itself:

Screen Shot 2019-09-22 at 8 56 33 PM

I tested this in both Firefox and Safari.

Verified that the other two are working now 👍

**Comment by codebykat** [comment]
Okay so for whatever reason it's not working to add it to `.note-list-item` in `theme.scss`, but it works if added to `.note-list-item` in `note-list/style.scss`.

**Pull request merged**